packages/teleport/src/components/MfaDeviceList/MfaDeviceList.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { useState, useMemo } from 'react';
+import styled from 'styled-components';
+import { sortBy } from 'lodash';
+import { ButtonBorder, Text } from 'design';
+import {
+  Column,
+  SortHeaderCell,
+  Cell,
+  TextCell,
+  SortTypes,
+  Table,
+} from 'design/DataTable';
+import { displayDate } from 'shared/services/loc';
+import { MfaDevice } from 'teleport/services/mfa/types';
+
+export default function MfaDeviceList({
+  devices = [],
+  onDelete,
+  isRecoveryFlow,
+}: Props) {
+  const [sortDir, setSortDir] = useState<Record<string, string>>({
+    registeredDate: SortTypes.ASC,
+  });
+
+  const mostRecentDevice = useMemo(() => sort(devices)[0], []);
```

Comment:
since it uses `useMemo` what happens to it when a new devices added?

Reply:
I have changed the implementation, `mostRecentDevice` is now going to be provided by the parent `Devices` component in Recovery Flow. 

It is not possible for new devices to be added on this step since this `mostRecentDevice` is only used when the device list is in the recovery flow (not the dashboard), it is set when the list of devices is retrieved from the backend.